feflow/protocols/nonequilibrium_cycling.py

```diff
@@ -350,16 +281,148 @@ def _execute(self, ctx, *, state_a, state_b, mapping, settings, **inputs):
 
-            # Equilibrate
-            context.setVelocitiesToTemperature(temperature)
+            # SERIALIZE SYSTEM, STATE, INTEGRATOR
+
+            system_ = context.getSystem()
+            state_ = context.getState(getPositions=True)
+            integrator_ = context.getIntegrator()
+
+            system_outfile = ctx.shared / 'system.xml.bz2'
+            state_outfile = ctx.shared / 'state.xml.bz2'
+            integrator_outfile = ctx.shared / 'integrator.xml.bz2'
+
+            serialize(system_, system_outfile)
+            serialize(state_, state_outfile)
+            serialize(integrator_, integrator_outfile)
+
+        finally:
+            # Explicit cleanup for GPU resources
+            del context, integrator
+
+        return {'system': system_outfile,
+                'state': state_outfile,
+                'integrator': integrator_outfile,
+                'phase': phase,
+                'initial_atom_indices': hybrid_factory.initial_atom_indices,
+                'final_atom_indices': hybrid_factory.final_atom_indices,
+                }
+
+
+class SimulationUnit(ProtocolUnit):
+    """
+    Monolithic unit for simulation. It runs NEQ switching simulation from chemical systems and stores the
+    work computed in numpy-formatted files, to be analyzed by another unit.
+    """
+
+    @staticmethod
+    def extract_positions(context, initial_atom_indices, final_atom_indices):
+        """
+        Extract positions from initial and final systems based from the hybrid topology.
+
+        Parameters
+        ----------
+        context: openmm.Context
+            Current simulation context where from extract positions.
+        hybrid_topology_factory: perses.annihilation.relative.HybridTopologyFactory
+            Hybrid topology factory where to extract positions and mapping information
+
+        Returns
+        -------
+
+        Notes
+        -----
+        It achieves this by taking the positions and indices from the initial and final states of
+        the transformation, and computing the overlap of these with the indices of the complete
+        hybrid topology, filtered by some mdtraj selection expression.
+
+        1. Get positions from context
+        2. Get topology from HTF (already mdtraj topology)
+        3. Merge that information into mdtraj.Trajectory
+        4. Filter positions for initial/final according to selection string
+        """
+        import numpy as np
+
+        # Get positions from current openmm context
+        positions = context.getState(getPositions=True).getPositions(asNumpy=True)
+
+        # Get indices for initial and final topologies in hybrid topology
+        initial_indices = np.asarray(initial_atom_indices)
+        final_indices = np.asarray(final_atom_indices)
+
+        initial_positions = positions[initial_indices, :]
+        final_positions = positions[final_indices, :]
+
+        return initial_positions, final_positions
```

Comment:
I think we want to maintain the capability of filtering the positions using an `mdtraj` selection, as it originally was in https://github.com/choderalab/feflow/blob/81e1b80ea762e0ea78bbde6fc5e68d774ceadb2b/feflow/protocols/nonequilibrium_cycling.py#L100 . Maybe we want to change the default to `all`? This does mean we would depend on `mdtraj` (maybe we can use `mdanalysis` for this if desired).

This should help reducing the storage used for trajectories when needed and might be useful when debugging/analyzing results, though I agree there are probably better ways to achieve this. One of them is creating the sampler (and reporter?) object in `openmmtools` for NEq cycling (analogue to what we do with `MultiStateSampler` for repex), and have that one handling all the trajectory/topologies serialization.

Are there any specific reasons other than not wanting to depend on `mdtraj`/`mdanalysis` that motivates this change?

Reply:
Note that the selection capabilities have are not being tested in CI. This is something that we would like to be testing if we decide to include it back again.

Reply:
Mainly I was motivated to not have to serialize the `hybrid_topology_factory`, or re-create it, in the `SimulationUnit`. So instead of passing that object around and into this method, we pass in just the positions. And since we don't have `hybridt_topology_factory`, we don't have `hybrid_topology`, so it's not so easy to do `mdtraj` manipulations anymore. But they seemed largely unnecessary?

I don't believe we use this method anywhere for trajectory storage or filtering. Isn't this method just used here, in this protocol? My view is that if we need do something different later, then we make it do that, but not pre-emptively.

Reply:
It just seemed like a feature that we weren't even using here, and its effect under default usage ("not water") was really nothing at all.

Reply:
Does this make sense? It's possible I've missed some important detail, so please correct me if so.

Reply:
The way this was being used is that when serializing trajectories we were extracting the positions and velocities of all things `not water`. It was just a way to not having to store them for everything and make the storage requirements a bit lighter, but it was used when storing the trajectories.

That said, I do think that the advantage of not having to pass the HTF object around might be a better compromise in this case, since the stored trajectories are not things we were using anyway. I think it's okay as it is now. If we end up needing to store only selections of atoms in the future we can give this another try.